adpcm-decoder/main.c

```diff
@@ -96,16 +96,16 @@ int main(int argc, char* argv[]) {
 
   //FIXME: Skip possible fact headers etc
 
-  char data_id[4];
-  fread(data_id, 4, 1, in);
-  if (strncmp(data_id, "data", 4) != 0) {
-    printf("Invalid DATA Id: %s\n", data_id);
+  char data_chunk_id[4];
+  fread(data_chunk_id, 4, 1, in);
+  if (strncmp(data_chunk_id, "data", 4) != 0) {
+    printf("Invalid DATA Id: %s\n", data_chunk_id);
```

Comment:
Should use the same text as the other chunk checks (also `%.4s` - obviously my original code was bad).

Hence my suggestion of moving it all in a function (but it feels like opening a can of worms; so just fix this text for now)

Please squash-up, too.

Reply:
i'll change that an squash it later!